### ISSUE TITLE
Introduce a logging interface, use structured logging

### ISSUE THREAD

**Issue by ncdc** [kcp-dev/kcp#221]
We should introduce interface such as go-logr and revamp all our logging to use it. Additionally, we should use structured logging for key-value pairs everywhere.

**Comment by ncdc** [comment]
@sttts I'd still like to embrace go-logr, but it's a pervasive code change (anywhere that we're logging). WDYT?

**Comment by avinal** [comment]
Hey @ncdc, 

I did some digging and found out that `runtime` package provides a [way to get filename and line number](https://pkg.go.dev/runtime#Caller). So maybe embracing `go-logr` results in breaking changes but we can always add some more text to the logs. Renaming the files is not the perfect solution. So here is what I am proposing. 

```go
func DebugInfo() string {
    _, file, line, _ := runtime.Caller(1)
    return file + ":" + strconv.Itoa(line)
}
```
This gives us output something like this:

```sh
I0311 23:08:07.631802  434779 file.go:47] /home/user/go/src/github.com/kcp-dev/kcp/cmd/kcp/file.go:47 <rest-of-the-message>
```

This way we don't have to rename file as well as not much changing the actual code file while getting the functionality we want. WDTY?

**Comment by marun** [comment]
I'm not sure it's desirable to require every single log call include additional detail like this.

**Comment by avinal** [comment]
> I'm not sure it's desirable to require every single log call include additional detail like this.

In that case, we just omit the extra information part from te klog logging functions. I guess we need details for errors, or anything other than just info.

**Comment by avinal** [comment]
Well, there is one more way, may seem too naive 😅 . we just add a constant to every file that stores the path from the project root to the file and pass it to the logging functions. This way there is no function calls just plain string and the rest of the information is already being printed. IIRC header guards in many C/C++ projects use a similar technique to make the header file unique.

**Comment by stevekuznetsov** [comment]
Implemented with #1694
